src/aten/sycl/BinaryKernels.cpp

```diff
@@ -42,7 +42,43 @@ struct DivFunctor {
   }
 };
 
-void add_kernel(TensorIterator& iter, const c10::Scalar& alpha) {
+template <typename scalar_t>
+struct RemainderIntegralFunctor {
+  scalar_t operator()(scalar_t a, scalar_t b) const {
+    scalar_t r = a % b;
+    if (r != 0 && c10::signs_differ(r, b)) {
+      r += b;
+    }
+    return r;
+  }
+};
+
+template <typename scalar_t>
+struct RemainderFloatingFunctor {
+  scalar_t operator()(scalar_t a, scalar_t b) const {
```

Comment:
Please check `__ubsan_ignore_float_divide_by_zero__`, which is used by both cpu and cuda kernels.